### PR TITLE
added support for modded rails

### PR DESCRIPTION
**Pull request by JonasJurczok**
Hi,

I added support for modded rails. In the config there is now a list of supported rail types. 
Wherever there was a check against the name there is now a check against the entity type.

This means all modded rails should work out of the box as they all share the `straight-rail` and `curved-rail` type.

Disclaimer: I only tested it on my machine in a heavily modded game with bioindustries powered rails.
Also I don't know your testing cycles.

### CODE REVIEW

**Comment by JonasJurczok** [comment]
As a side note:
One thing I read multiple times is that you should put all your global tables into a namespace.
The global table is shared between all mods. If there is one that uses a table with the same name you can end up in deep doodoo. Maybe move everything to `global.releltrains` or something :)

File: Realistic_Electric_Trains_0.1.1/control.lua [hunk 1]
```diff
@@ -6,14 +6,14 @@ require("logic.overhead_line")
...
-		global.electric_locos = {}  -- Loco ID -> Loco Entity
-	end
+		function(e)
+			-- init lookup tables
+			global.wire_for_pole = {}   -- Pole ID -> Wire Entity
+			global.power_for_pole = {}  -- Pole ID -> Power Entity
+			global.graphic_for_pole = {}-- Pole ID -> Graphic Entity
+			global.power_for_rail = {}  -- Rail ID -> Power Entity
+			global.electric_locos = {}  -- Loco ID -> Loco Entity
+		end
```
Comment (JonasJurczok) [inline Realistic_Electric_Trains_0.1.1/control.lua]: Sorry about that.. the IDE's auto indentation algorithm thought this is more correct. If you don't like it tell me and I revert that.

File: Realistic_Electric_Trains_0.1.1/control.lua [hunk 2]
```diff
@@ -218,36 +220,34 @@ do
...
+					power_provider.energy - transfer
 				end
 			end
 		end
 	end
 
 end
 
-script.on_event(defines.events.on_tick, on_tick)
+script.on_nth_tick(10, on_tick)
```
Comment (JonasJurczok) [inline Realistic_Electric_Trains_0.1.1/control.lua]: This is the performance change.

**Comment by Scarabytes** [comment]
Was merged (partially) into the new version.

**Comment by JonasJurczok** [comment]
Awesome, thanks.

**Pull request closed without merging**